### PR TITLE
[NOCSL] Fix failing tests

### PR DESCRIPTION
**Pull request by Mudaafi**
Updating our SDK tests to match backend endpoints updates.

### CODE REVIEW

File: tests/modules/test_catalog_csv.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def test_with_replace_items():
 
     response = catalog.replace_catalog(data)
 
-    assert isinstance(response.get('task_id'), int)
+    assert isinstance(response.get('task_id'), str)
```
Comment (Mudaafi) [inline tests/modules/test_catalog_csv.py:41]: Backend returns the task_id as a string for `replace_catalog`

File: tests/modules/test_catalog_csv.py [hunk 2]
```diff
@@ -95,7 +95,7 @@ def test_with_replace_no_files():
 
     with raises(
         HttpException,
-        match=r'form-data should contain at least one of "items", "variations", "item_groups'
+        match=r"Something unexpected went wrong here. We're looking into it. Sorry for the inconvenience!"
```
Comment (Mudaafi) [inline tests/modules/test_catalog_csv.py:98]: Might need to raise this to backend.

File: tests/modules/test_recommendations.py [hunk 3]
```diff
@@ -264,7 +264,7 @@ def test_with_valid_pod_id_and_filters():
     assert isinstance(response.get('request'), dict)
     assert isinstance(response.get('response'), dict)
     assert isinstance(response.get('result_id'), str)
-    assert response.get('request').get('filters').get('keywords') == filters['keywords'][0]
+    assert response.get('request').get('filters').get('keywords') == filters['keywords']
```
Comment (Mudaafi) [inline tests/modules/test_recommendations.py:267]: Response returned is an array even if it's a single keyword.

File: tests/modules/test_tasks.py [hunk 4]
```diff
@@ -40,9 +40,9 @@ def test_get_all_tasks_with_no_params():
     assert isinstance(response.get('total_count'), int)
     assert len(response.get('tasks')) <= 20 and len(response.get('tasks')) >= 1
 
-    task = next(filter(lambda task: task.get('id') == task_id_no_params_test, response.get('tasks')), None)
+    task = next(filter(lambda task: str(task.get('id')) == task_id_no_params_test, response.get('tasks')), None)
     assert isinstance(task, dict)
-    assert task.get('id') == task_id_no_params_test
+    assert str(task.get('id')) == task_id_no_params_test
```
Comment (Mudaafi) [inline tests/modules/test_tasks.py:45]: Tasks endpoint returns `task_id` as an int whereas the Catalog endpoint returns it as a string. See line 25.

File: tests/modules/test_catalog_csv.py (hunk 2 above)
Comment (esezen) [reply tests/modules/test_catalog_csv.py:98]: @Mudaafi Let's raise this with backend.

File: tests/modules/test_tasks.py (hunk 4 above)
Comment (esezen) [reply tests/modules/test_tasks.py:45]: Can we also raise this with backend?

**Review by esezen — APPROVED** [review]
This is looking good. Can we hold on merging before we talk to backend if this is not blocking anything? If possible, catalog and tasks API should both return string for task ID. Maybe they can change it in the backend instead of us changing it here.

File: tests/modules/test_catalog_csv.py (hunk 1 above)
Comment (Mudaafi) [reply tests/modules/test_catalog_csv.py]: Fix deployed to backend.

File: tests/modules/test_catalog_csv.py (hunk 2 above)
Comment (Mudaafi) [reply tests/modules/test_catalog_csv.py]: Fix has been deployed to backend so I've reverted the changes.

File: tests/modules/test_tasks.py (hunk 4 above)
Comment (Mudaafi) [reply tests/modules/test_tasks.py]: I've checked and they all endpoints return ints. Reverted the changes.

**Comment by Mudaafi** [comment]
We should be good to merge this now

**Pull request merged**